### PR TITLE
[JENKINS-55227] Extends debug documentation

### PR DESCRIPTION
**Pull request by varyvol**
[JENKINS-55227](https://issues.jenkins-ci.org/browse/JENKINS-55227)

Extends the documentation for debug.

### CODE REVIEW

**Review by fcojfernandez — APPROVED** [review]

File: DOCKER.md [hunk 1]
```diff
@@ -33,10 +33,14 @@ root@dec4c0f12478:/src# /app/bin/jenkinsfile-runner -w /tmp/jenkins -p /usr/shar
...
-Then, set the `DEBUG` environment variable and expose the port where to connect the remote debug
+Then, set the `DEBUG` environment variable and expose the port where to connect the remote debug. Note Jenkinsfile Runner itself
+considers 5005 as debugging port but you can map such port to whatever value you prefer through Docker port mapping.
 
 ```bash
 docker run --rm -e DEBUG=true -p 5005:5005 -v $PWD/test:/workspace jenkinsfile-runner:my-production-jenkins
 ```
+
+In case you are having issues when the Docker image is generated in another way (for example through [Custom WAR Packager](https://github.com/jenkinsci/custom-war-packager/)),
+you can simulate the configuration there (plugin versions, WAR file, ...) through the Vanilla image and then debug.
```
Comment (oleg-nenashev) [inline DOCKER.md]: I would recommend just adding the JAVA_OPTS command sample here

Comment (varyvol) [reply DOCKER.md]: In theory you'll want to debug when you are developing Jenkinsfile Runner or having any issues. Simulating a configuration (for example a Docker image generated through Custom War Packager) should be possible through the plugins.txt file and some configuration, so no need to use the JAVA_OPTS option I think (or at least mentioning it here).

Comment (oleg-nenashev) [reply DOCKER.md]:
```suggestion
you can directly pass `JAVA_OPTS` to the Docker run arguments:

```bash
docker run --rm -e JAVA_OPTS='-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005' -p 5005:5005 -v $PWD/test:/workspace jenkinsfile-runner:my-production-jenkins
```

```

Comment (oleg-nenashev) [reply DOCKER.md]: Rendered like 💩 , but I hope it represents what I meant

**Review by oleg-nenashev — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
```
[2019-03-18T16:49:38.364Z] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-install-plugin:2.5.2:install (default-install) on project parent: Failed to install artifact io.jenkins.jenkinsfile-runner:parent:pom:1.0-beta-8-SNAPSHOT: /home/jenkins/.m2/repository/io/jenkins/jenkinsfile-runner/parent/1.0-beta-8-SNAPSHOT/parent-1.0-beta-8-SNAPSHOT.pom (No such file or directory) -> [Help 1]
```

**Pull request merged**